tensorboard/data/proto/data_provider.proto

```diff
@@ -31,6 +35,38 @@ service TensorBoardDataProvider {
   rpc ReadBlob(ReadBlobRequest) returns (stream ReadBlobResponse) {}
 }
 
+message GetExperimentRequest {
+  string experiment_id = 1;
+}
+
+message GetExperimentResponse {
+  // Human-readable description of the data source.
+  //
+  // This might be an absolute path on disk to a log directory of event files,
+  // a relative path, a URL to a cloud storage bucket, an address for a local
+  // socket server, etc. There does not need to be any way to "dereference" this
+  // address to read the original data directly.
+  string data_location = 1;
+
+  // User-facing experiment name.
+  //
+  // This name is typically set by a user explicitly or automatically set by a
+  // training system.
+  string name = 2;
+
+  // User-facing experiment description.
+  string description = 3;
+
+  // Time that the experiment was created.
+  //
+  // This does not necessarily have any relation to times of events within an
+  // experiment. A new experiment may contain old data, or data may be freshly
+  // written to an experiment after it is created.
+  //
+  // May be unset if no creation time is known.
+  google.protobuf.Timestamp creation_time = 4;
```

Comment:
@nfelt: Could you take a quick look at this use of `Timestamp` in light
of the build changes in this PR and its diffbase, #4749? I recall that
in #4314 you lobbied for using `Timestamp` rather than `double`, and in
this case I agree that it’s the right tradeoff as far as the proto
definition goes. It incurs a bit of build complexity; IMHO it’s worth
it, but I would also not mind just using `double` if we prefer that.

Reply:
I'm ok with the build complexity since it seems fairly contained, although boy do I wish `com_google_protobuf` were a nicer structured Bazel dependency to have.